tools/rvn/CommandLineApp.cs

```diff
@@ -51,6 +61,137 @@ public static int Run(string[] args)
+        {
+            byte[] zipFile;
+            SetupInfo setupInfo;
+
+            if (File.Exists(parameters.SetupInfoPath) == false)
+                return ExitWithError($"{nameof(parameters.SetupInfoPath)} does not exits", parameters.Command);
+
+            try
+            {
+                using (StreamReader file = File.OpenText(parameters.SetupInfoPath))
+                {
+                    JsonSerializer serializer = new();
+                    setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+                }
+            }
+            catch (Exception e)
+            {
+                return ExitWithError($"Failed to deserialize SetupInfo object form this file path {nameof(parameters.SetupInfoPath)}\nError: {e}",parameters.Command);
+            }
+
+            if (string.IsNullOrEmpty(setupInfo?.Domain))
+                return ExitWithError($"{nameof(setupInfo.Domain)} cannot be empty",parameters.Command);
+            
+            parameters.PackageOutputPath ??= setupInfo.Domain;
+            
+            if (string.IsNullOrEmpty(parameters.CertPassword.PassFromEnv) == false)
+            {
+                setupInfo.Password = parameters.CertPassword.PassFromEnv;
+            }
+            else if (string.IsNullOrEmpty(parameters.CertPassword.passFromCmd) == false)
+            {
+                setupInfo.Password = parameters.CertPassword.PassFromEnv;
+            }
+
+            switch (parameters.SetupMode)
+            {
+                case ImportCertificateMode when string.IsNullOrEmpty(parameters.CertificatePath):
+                {
+                    return ExitWithError("--path option must be set", parameters.Command);
+                }
+                case ImportCertificateMode:
+                {
+                    
+                    var certBytes = await File.ReadAllBytesAsync(parameters.CertificatePath, parameters.CancellationToken);
+                    var certBase64 = Convert.ToBase64String(certBytes);
+                    setupInfo.Certificate = certBase64;
+                    zipFile = await LetsEncryptUtils.ImportCertificateSetup(setupInfo, parameters.Progress, parameters.CancellationToken);
+                    break;
+                }
+                case LetsEncryptMode when string.IsNullOrEmpty(parameters.SetupInfoPath) == false:
+                {
+                    zipFile = await LetsEncryptUtils.SetupLetsEncrypt(setupInfo, parameters.Progress, parameters.CancellationToken);
+                    break;
+                }
+                default:
+                    return ExitWithError("--mode and --setup-params-path options must be set.", parameters.Command);
+            }
+
+            try
+            {
+                if (Path.GetExtension(parameters.PackageOutputPath)?.Equals(".zip") == false)
+                    return ExitWithError($"'{parameters.PackageOutputPath}' file name must end with ZIP extenstion", parameters.Command);
```

Comment:
User would probably like to know that before the whole behemoth getting the lets encrypt certificate, contacting api and generating settings kicks off :)

Reply:
you are right